From 0f03291859d091422657da793da12f9bda53ec56 Mon Sep 17 00:00:00 2001
From: Lim Siew Hoon <siew.hoon.lim@intel.com>
Date: Tue, 18 May 2021 15:44:26 +0800
Subject: [PATCH] glimagesink: rebased downrank to marginal

From 79841a02ee754eba736cb32aaf2ed2b2fc0483d4 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Thu, 24 Sep 2015 19:47:32 +0300
Subject: [PATCH 4/4] glimagesink: Downrank to marginal

On desktop, where there is good OpenGL, xvimagesink will come up first,
on other platforms, OpenGL can't be trusted because it's either software (like
in a VM) or broken (like on embedded)., so let ximagesink come above.

Upstream-Status: Submitted [https://bugzilla.gnome.org/show_bug.cgi?id=751684]

Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 ext/gl/gstglimagesink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/gl/gstglimagesink.c b/ext/gl/gstglimagesink.c
index aca393252..fcd834346 100644
--- a/ext/gl/gstglimagesink.c
+++ b/ext/gl/gstglimagesink.c
@@ -111,7 +111,7 @@ typedef GstGLSinkBinClass GstGLImageSinkBinClass;
 
 G_DEFINE_TYPE (GstGLImageSinkBin, gst_gl_image_sink_bin, GST_TYPE_GL_SINK_BIN);
 GST_ELEMENT_REGISTER_DEFINE_WITH_CODE (glimagesink, "glimagesink",
-    GST_RANK_SECONDARY, gst_gl_image_sink_bin_get_type (),
+    GST_RANK_MARGINAL, gst_gl_image_sink_bin_get_type (),
     gl_element_init (plugin));
 
 enum
-- 
2.31.1

